packages/react-make-styles/src/makeStyles.ts

```
@@ -1,12 +1,34 @@
 import { makeStyles as vanillaMakeStyles, MakeStylesOptions, MakeStylesStyleRule } from '@fluentui/make-styles';
 import { useFluent } from '@fluentui/react-shared-contexts';
 import { Theme } from '@fluentui/react-theme';
+import * as React from 'react';
 
 import { useRenderer } from './RendererContext';
 
+function isInsideComponent() {
+  try {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    React.useContext(({} as unknown) as React.Context<unknown>);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
 export function makeStyles<Slots extends string>(stylesBySlots: Record<Slots, MakeStylesStyleRule<Theme>>) {
   const getStyles = vanillaMakeStyles(stylesBySlots);
 
+  if (process.env.NODE_ENV !== 'production') {
+    if (isInsideComponent()) {
+      throw new Error(
```

I totally agree with throwing on errors like this, but that's definetely not the approach being used at the moment. We're not throwing on missuse, shouldn't this just warn the user?

---

I would prefer to log an error i.e. `console.error()`, but:
- it's a critical missuse 
- users are ignoring errors until it's really breaks something ¯\_(ツ)_/¯

---

Discussed offline with the team, all vote to throw.